src/main/recorder.ts

```diff
@@ -192,58 +244,52 @@ type RecorderOptionsType = {
                 console.info("[Recorder] Raid encounter was too short, discarding");
+            } else if (discardVideo) {
+                console.info("[Recorder] Discarding video");
+            } else {
+                const bufferFile = obsRecorder.getObsLastRecording();
+                if (bufferFile) {
+                    await this.queueVideo(bufferFile, metadata);
+                } else {
+                    console.error("[Recorder] Unable to get the last recording from OBS. Can't process video.");
+                }
             }
 
-            // Run the size monitor to ensure we stay within size limit.
-            // Need some maths to convert GB to bytes
-            runSizeMonitor(this._options.storageDir, this._options.maxStorage)
-                .then(() => {
-                    if (mainWindow) mainWindow.webContents.send('refreshState');
-                });
-
-            // Clean-up the temporary recording directory. 
-            this.cleanupBuffer(1);
-
             // Refresh the GUI
             if (mainWindow) mainWindow.webContents.send('refreshState');
 
             // Restart the buffer recording ready for next game.
-            setTimeout(async () => {
-                this.startBuffer();
-            }, 5000)
+            this.startBuffer();
         }, 
         overrun * 1000);
     }
 
     /**
-     * Finalize the video by cutting it to size, moving it to the persistent
-     * storage directory and writing the metadata JSON file. 
+     * Queue the video for processing.
      * 
      * @param {Metadata} metadata the details of the recording
      */
-    finalizeVideo = async (metadata: Metadata, outputFilename?: string): Promise<string> => {
-
-        // Gnarly syntax to await for the setTimeout to finish.
-        return new Promise<string> ((resolve) => {
-
+    queueVideo = async (bufferFile: string, metadata: Metadata): Promise<void> => {
+        return new Promise<void>((resolve) => {
             // It's a bit hacky that we async wait for 2 seconds for OBS to 
             // finish up with the video file. Maybe this can be done better. 
             setTimeout(async () => {
-                const bufferedVideo = await getNewestVideo(this._options.bufferStorageDir);
-                const videoPath = await cutVideo(bufferedVideo, this._options.storageDir, outputFilename, metadata.duration);
-                await writeMetadataFile(videoPath, metadata);
-                console.log('[Recorder] Finalized video', videoPath);
-                resolve(videoPath);
+                const queueItem: VideoQueueItem = {
+                    bufferFile,
+                    metadata,
+                };
+
+                console.log("[Recorder] Queuing video for processing", queueItem);
+
+                this._videoQueue.write(queueItem);
+                resolve();
+
                 if (mainWindow) mainWindow.webContents.send('updateStatus', AppStatus.ReadyToRecord);
```

Comment:
I've no idea why this was already like this, but surely this should come before the resolve?